irc_callback.go

```diff
@@ -232,33 +232,88 @@ func (irc *Connection) setupCallbacks() {
 	// work. It has to be set somewhere first in case the nick is already
 	// taken or unavailable from the beginning.
 	irc.AddCallback("437", func(e *Event) {
-		// If irc.nickcurrent hasn't been set yet, set to irc.nick
-		if irc.nickcurrent == "" {
-			irc.nickcurrent = irc.nick
+		irc.Lock()
+		defer irc.Unlock()
+		if !irc.fully_connected {
+			if irc.nickcurrent == "" {
+				irc.nickcurrent = irc.nick
+			}
+			if len(irc.nickcurrent) > 8 {
+				irc.nickcurrent = "_" + irc.nickcurrent
+			} else {
+				irc.nickcurrent = irc.nickcurrent + "_"
+			}
+			irc.SendRawf("NICK %s", irc.nickcurrent)
 		}
+	})
 
-		if len(irc.nickcurrent) > 8 {
-			irc.nickcurrent = "_" + irc.nickcurrent
-		} else {
-			irc.nickcurrent = irc.nickcurrent + "_"
+	irc.AddCallback("433", func(e *Event) {
+		irc.Lock()
+		defer irc.Unlock()
+		if !irc.fully_connected {
+			if irc.nickcurrent == "" {
+				irc.nickcurrent = irc.nick
+			}
+			if len(irc.nickcurrent) > 8 {
+				irc.nickcurrent = "_" + irc.nickcurrent
+			} else {
+				irc.nickcurrent = irc.nickcurrent + "_"
+			}
+			irc.SendRawf("NICK %s", irc.nickcurrent)
 		}
-		irc.SendRawf("NICK %s", irc.nickcurrent)
 	})
 
-	// 433: ERR_NICKNAMEINUSE "<nick> :Nickname is already in use"
-	// Add a _ to current nick.
-	irc.AddCallback("433", func(e *Event) {
-		// If irc.nickcurrent hasn't been set yet, set to irc.nick
-		if irc.nickcurrent == "" {
-			irc.nickcurrent = irc.nick
+	irc.AddCallback("431", func(e *Event) {
+		irc.Lock()
+		defer irc.Unlock()
+		if !irc.fully_connected {
+			if irc.nickcurrent == "" {
+				irc.nickcurrent = irc.nick
+			}
+			if len(irc.nickcurrent) > 8 {
+				irc.nickcurrent = "_" + irc.nickcurrent
+			} else {
+				irc.nickcurrent = irc.nickcurrent + "_"
+			}
+			irc.SendRawf("NICK %s", irc.nickcurrent)
 		}
+	})
 
-		if len(irc.nickcurrent) > 8 {
-			irc.nickcurrent = "_" + irc.nickcurrent
-		} else {
-			irc.nickcurrent = irc.nickcurrent + "_"
+	irc.AddCallback("432", func(e *Event) {
+		irc.Lock()
+		defer irc.Unlock()
+		if !irc.fully_connected {
+			if irc.nickcurrent == "" {
+				irc.nickcurrent = irc.nick
+			}
+			// Dodajemy prefiks 'Err' aby spróbować inny pseudonim
+			irc.nickcurrent = "Err" + irc.nickcurrent
+			irc.SendRawf("NICK %s", irc.nickcurrent)
+		}
+	})
+
+	irc.AddCallback("436", func(e *Event) {
+		irc.Lock()
+		defer irc.Unlock()
+		if !irc.fully_connected {
+			if irc.nickcurrent == "" {
+				irc.nickcurrent = irc.nick
+			}
+			if len(irc.nickcurrent) > 8 {
+				irc.nickcurrent = "_" + irc.nickcurrent
+			} else {
+				irc.nickcurrent = irc.nickcurrent + "_"
+			}
+			irc.SendRawf("NICK %s", irc.nickcurrent)
+		}
+	})
+
+	irc.AddCallback("484", func(e *Event) {
+		irc.Lock()
+		defer irc.Unlock()
+		if !irc.fully_connected {
+			// Zachowaj obecny nick i nie podejmuj kolejnej próby zmiany
```

Comment:
english

Reply:
Will be fixed also, like I said before go is new for me (working with git pull/push also :P)